libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp

```diff
@@ -539,29 +539,45 @@ void AC_AttitudeControl_Heli::set_throttle_out(float throttle_in, bool apply_ang
     _throttle_in = throttle_in;
     update_althold_lean_angle_max(throttle_in);
 
-    if (_transition_count > 0) {
-        _transition_count -= 1;
+    _motors.set_throttle_filter_cutoff(filter_cutoff);
+    if (apply_angle_boost) {
+        // Apply angle boost
+        throttle_in = get_throttle_boosted(throttle_in);
     } else {
-        _transition_count = 0;
+        // Clear angle_boost for logging purposes
+        _angle_boost = 0.0f;
     }
-    float throttle_out = 0.0f;
-    if (_transition_count > 0) {
-        if ((_ahrs.roll_sensor >= -3000 && _ahrs.roll_sensor <= 3000) || _ahrs.roll_sensor >= 15000 || _ahrs.roll_sensor <= -15000) {
-            throttle_out = (throttle_in - ((AP_MotorsHeli&)_motors).get_coll_mid()) / cosf(radians(_ahrs.roll_sensor * 0.01f)) + ((AP_MotorsHeli&)_motors).get_coll_mid();
-        } else if ((_ahrs.roll_sensor > 3000 && _ahrs.roll_sensor < 15000) || (_ahrs.roll_sensor > -15000 && _ahrs.roll_sensor < -3000)) {
-            float scale_factor = cosf(radians(_ahrs.roll_sensor * 0.01f)) / cosf(radians(30.0f));
-            throttle_out = scale_factor * (throttle_in - ((AP_MotorsHeli&)_motors).get_coll_mid())/ cosf(radians(30.0f)) + ((AP_MotorsHeli&)_motors).get_coll_mid();
-        }
-    } else if (_inverted_flight) {
-        throttle_out = 1.0f - throttle_in;
-    } else {
-        throttle_out = throttle_in;
+    _motors.set_throttle(throttle_in);
+}
```

Comment:
I think you can just remove the override now. The only difference is the base class also calls `_motors.set_throttle_avg_max(get_throttle_avg_max(MAX(throttle_in, _throttle_in)));` heli motors doesn't use this value so it should not matter that were now setting it when we weren't before.

Reply:
@IamPete1 So you are saying that I should move the multi set_throttle_out to the base class and remove the functions from multi and heli since they can be the same?